tfexec/main.go

```diff
@@ -0,0 +1,27 @@
+package tfexec
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+// FindTerraform attempts to find a Terraform CLI executable.
+//
+// As a first preference it will look for the environment variable
+// TFEXEC_TERRAFORM_PATH and return its value. If that variable is not set, it will
+// look in PATH for a program named "terraform",
+// and, if one is found, return its absolute path.
+func FindTerraform() (string, error) {
+	if p := os.Getenv("TFEXEC_TERRAFORM_PATH"); p != "" {
```

Comment:
I don't know if this package should read environment, people can always pass it in.

Reply:
We may just want to restructure the binary finder/install to a strategy pattern or something, ie. look path, explicit path, env vars, install from releases, are all just slightly different strategies for this and may be combined, etc.